src/std_io.py

```diff
@@ -0,0 +1,167 @@
+
+            if not state.enableNetwork and state.enableGUI:
+                # in airgap mode, trick the status icon that we are in fact
+                #   NOT waiting for a connection
+                # (may be removed after impact analysis)
+                shared.statusIconColor = 'yellow'
+
+            if stdInputMode == 'airgap':
+                # airgap mode uses the timestamp
+                # unhex the timestamp with error rejection
+                if len(hexTStamp) == 16:
+                    try:
+                        # stdioStamp, = unpack('>Q', unhexlify(hexTStamp))
+                        _, = unpack('>Q', unhexlify(hexTStamp))
+                    except Exception:
+                        logger.info("STDIN: Invalid timestamp format: " + hexTStamp)
+                        continue
+
+            # check that proof of work is sufficient.
+            if not protocol.isProofOfWorkSufficient(binObject):
+                logger.info('STDIN: Insufficient Proof of Work')
+                continue
+
+            # extract expiry time, object type
+            eTime, = unpack('>Q', binObject[8:16])
+            objectType, = unpack('>I', binObject[16:20])
+
+            # extract version number and stream number
+            readPosition = 20  # bypass the nonce, time, and object type
+            # versionNumber, versionLength
+            _, versionLength = decodeVarint(binObject[readPosition:readPosition + 10])
+            readPosition += versionLength
+            streamNumber, streamNumberLength = decodeVarint(binObject[readPosition:readPosition + 10])
+            readPosition += streamNumberLength
+
+            # calculate inventory hash
+            inventoryHash = calculateInventoryHash(binObject)
+
+            # duplicate check on inventory hash (both netcat and airgap)
+            if inventoryHash in Inventory():
+                logger.info("STDIN: Already got object " + hexlify(inventoryHash))
+                continue
+
+            # in netcat mode, push object to inventory and id to output queue
+            if stdInputMode == 'netcat':
+                Inventory()[inventoryHash] = (objectType, streamNumber, binObject, eTime, '')
+                logger.info("STDIN: Accepted object (type=%u) " % objectType + hexlify(inventoryHash))
+                queues.invQueue.put((streamNumber, inventoryHash))
+
+            # honour global shutdown flag
+            if state.shutdown != 0:
+                logger.info('stdInput thread shutdown.')
+                break
+
+
+class objectStdOut(threading.Thread):
+    """
+    The objectStdOut thread receives network objects from the receiveDataThreads.
+    """
+    def __init__(self):
+        threading.Thread.__init__(self, name="objectStdOut")
+
+        # REFACTOR THIS with objectProcessor into objectProcessorQueue
+        queryreturn = sqlQuery(
+            '''SELECT objecttype, data FROM objectprocessorqueue''')
+        for row in queryreturn:
```

Comment:
By waiting until we have synchronously pulled all objects from the database and put all objects on the queue we are missing out on some benefits of parallelism and we'll hit a memory limit with large data sets. Maybe this is unavoidable or related to your suggested refactoring.

Also, keeping the number of places where raw SQL is used to a minimum is a good idea. Again, I assume this would be part of your suggested refactoring.